float32 velocity
float32 distance
float32 angle
uint32 new_mesg
unint8 blocking